### DISCUSSION TITLE
Struggling with starting any server below 1.18

### DISCUSSION THREAD

**Discussion by BurntChikken** [itzg/docker-minecraft-server#2597] [Q&A] [answered]
Just to begin I am very much new to these things, being github and docker so let me know if any of what I say doesn't make sense.

I am struggling when it comes to composing the server, as when I do so it comes up with the below logs. Tried specifying different versions of forge, which I grabbed from curseforge on my PC, which opens fine. It seems to be some kind of issue with forge or something is my guess because it works on 1.18 and above, 1.16 and 1.17 apparently don't support forge, anything below these comes up with the same error code etc. I will also link my docker-compose.yml file below.

>Logs;
xxx@xxxxxxx:~/Containers/mcj_server$ sudo docker-compose up
Recreating mcj_server_mc_1 ... done
Attaching to mcj_server_mc_1
mc_1  | [init] Changing ownership of /data to 1000 ...
mc_1  | [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 2 1000 1000 4096 Jan 16 05:33 /data'
mc_1  | [init] Resolving type given FORGE
mc_1  | [mc-image-helper] 05:33:38.339 INFO  : Downloading Forge installer 14.21.1.2387 for Minecraft 1.12
mc_1  | [mc-image-helper] 05:33:44.059 INFO  : Running Forge 14.21.1.2387 installer for Minecraft 1.12. This might take a while...
mc_1  | [init] Creating server properties in /data/server.properties
mc_1  | [init] Disabling whitelist functionality
mc_1  | [mc-image-helper] 05:34:15.339 INFO  : Created/updated 4 properties in /data/server.properties
mc_1  | [init] Setting initial memory to 4G and max to 4G
mc_1  | [init] Starting the Minecraft server...
mc_1  | A problem occurred running the Server launcher.java.lang.reflect.InvocationTargetException
mc_1  |         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
mc_1  |         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
mc_1  |         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
mc_1  |         at java.base/java.lang.reflect.Method.invoke(Unknown Source)
mc_1  |         at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:62)
mc_1  |         at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:31)
mc_1  | Caused by: java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
mc_1  |         at net.minecraft.launchwrapper.Launch.<init>(Launch.java:34)
mc_1  |         at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
mc_1  |         ... 6 more
mc_1  | 2024-01-16T05:34:19.600Z        WARN    mc-server-runner        Minecraft server failed. Inspect logs above for errors that indicate cause. DO NOT report this line as an error.        {"exitCode": 1}
mc_1  | 2024-01-16T05:34:19.600Z        INFO    mc-server-runner        Done
mcj_server_mc_1 exited with code 1

>docker-compose.yml;
version: "3.8"
services:
  mc:
    image: itzg/minecraft-server
    tty: true
    stdin_open: true
    ports:
      - "25565:25565"
    environment:
      TYPE : FORGE
      MEMORY : 4G
      VERSION : "1.12"
      FORGE_VERSION : "14.21.1.2387"
      EULA: "TRUE"
    volumes:
      # attach the relative directory 'data' to the container's /data path
      - ./data:/data

**Comment by itzg** [comment, marked as answer]
Use a Java 8 image tag as described by the note here

https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/
